deku-derive/src/macros/deku_read.rs

```diff
@@ -352,6 +364,24 @@ fn emit_enum(input: &DekuData) -> Result<TokenStream, syn::Error> {
         });
     }
 
+    // create `deku_id()` function for retrieving at runtime the id value
+    let deku_id_id_type = if let Some(id_type) = id_type {
+        id_type
+    } else {
+        &ctx_types
+    };
+    let deku_id = quote! {
+        impl #ident {
```

Comment:
I think a `DekuEnumExt` trait to hold these functions would be great as @passcod mentioned